lib/parser.js

```diff
@@ -74,39 +80,17 @@ async function parse(asyncapiYAMLorJSON, options = {}) {
 
     if (options.applyTraits === undefined) options.applyTraits = true;
 
-    try {
-      parsedJSON = await $RefParser.dereference(options.path, parsedJSON, {
-        parse: options.parse,
-        resolve: options.resolve,
-        dereference: options.dereference,
-      });
-    } catch (err) {
-      throw new ParserError({
-        type: 'dereference-error',
-        title: err.message,
-        parsedJSON,
-        refs: findRefs(parsedJSON, err.originalPath || err.path, options.path, initialFormat, asyncapiYAMLorJSON),
-      });
-    }
-  } catch (e) {
-    if (e instanceof ParserError) throw e;
-    throw new ParserError({
-      type: 'unexpected-error',
-      title: e.message,
-      parsedJSON,
-    });
-  }
-
-  const ajv = new Ajv({
-    jsonPointers: true,
-    allErrors: true,
-    schemaId: 'id',
-    logger: false,
-  });
-
-  ajv.addMetaSchema(require('ajv/lib/refs/json-schema-draft-04.json'));
+    await dereference(parsedJSON, initialFormat, asyncapiYAMLorJSON, options);
```

Comment:
Following my comment above (saying `dereference` shouldn't be an option anymore):

```suggestion
    await dereference(parsedJSON, initialFormat, asyncapiYAMLorJSON, { ...options, dereference: 'ignore' });
```